<html><body class="markdown-body">
<meta http-equiv="content-type" content="text/html; charset=utf-8"/>
<link rel="stylesheet" href="css/style.css"/>
<script src="js/jquery-1.10.1.js"></script>
<script src="js/doc.js"></script>
<p><a href="lang">言語仕様</a></p>
<h1 id="-">例外</h1>
<p>実行中に何らかの問題が発生した場合には例外が発生します．例えば，次のようなものが考えられます．</p>
<ul>
<li><code>null</code> や <code>undefined</code> に対するフィールドの読み出しを行う</li>
<li>ファイルの読み出しを行うときに該当する<a href="file">ファイル</a>が存在しない</li>
<li>無限ループが発生している</li>
<li>一度に大量のオブジェクトを作成した</li>
</ul>
<h2 id="-">例外を検知する</h2>
<p>例外が発生しそうな場所をtry ... catch で囲むと，例外が発生したときに処理を行うことができます．</p>
<pre><code class="lang-JS">try {
    a=null;
    print(a.hoge);
} catch (e) {
    print(&quot;例外が発生しました&quot;,e);
}
</code></pre>
<h2 id="-">例外を明示的に発生させる</h2>
<p><code>throw e</code>で例外を発生させることができます．<em>e</em>には<code>Error</code>オブジェクトを指定します．</p>
<pre><code class="lang-JS">try {
    if (a==null) {
        throw new Error(&quot;a に値が設定されていません．&quot;);
    }
    print(a.hoge);
} catch (e) {
    print(&quot;例外が発生しました&quot;,e);
}
</code></pre>
</body></html>
